### PR TITLE
MacOS simple support

### PR DESCRIPTION
**Pull request by LifeKILLED**
image

Include OpenGL 3.0 with extensions for MacOS, incompatible code fixed by macros. Tested on MacBook Air 2020 (M1)

Also need to update OpenGL library with this merge request: https://github.com/Panzerschrek/panzer_ogl_lib/pull/4

### CODE REVIEW

File: src/system_window.cpp [hunk 1]
```diff
@@ -568,12 +572,14 @@ void SystemWindow::EndFrame()
 
 		glEnable( GL_TEXTURE_2D );
 
+		#ifndef __APPLE__
 		glBegin( GL_QUADS );
```
Comment (Panzerschrek) [inline src/system_window.cpp]: It seems like MacOS doesn't support OpenGL compatibility profile, only core profile, where `glBegin`-like function do not exist anymore. Am i right?

File: src/system_window.cpp [hunk 2]
```diff
@@ -256,8 +256,10 @@ SystemWindow::SystemWindow( Settings& settings )
 	viewport_size_.Width ()= static_cast<unsigned int>( width  / scale );
 	viewport_size_.Height()= static_cast<unsigned int>( height / scale );
 
+	#ifndef __APPLE__
 	use_gl_context_for_software_renderer_= !is_opengl && settings_.GetOrSetBool( "r_software_use_gl_screen_update", true );
-	if( is_opengl )
+	#endif
+    if( is_opengl )
```
Comment (Panzerschrek) [inline src/system_window.cpp]: Remove unnecessary whitespace changes in this line (and maybe in other lines).

File: src/system_window.cpp [hunk 3]
```diff
@@ -568,12 +572,14 @@ void SystemWindow::EndFrame()
 
 		glEnable( GL_TEXTURE_2D );
 
+		#ifndef __APPLE__
```
Comment (Panzerschrek) [inline src/system_window.cpp]: I suggest to wrap the whole branch with `use_gl_context_for_software_renderer_` into conditional compilation.
```cpp
else if( use_gl_context_for_software_renderer_ )
{
#ifndef __APPLE__
...
#endif
}```

**Review by Panzerschrek — CHANGES_REQUESTED** [review]

**Comment by Panzerschrek** [comment]
It's a surprise for me that so little changes are needed to create MacOS port. I never kept this OS in mind.

**Comment by LifeKILLED** [comment]
> It's a surprise for me that so little changes are needed to create MacOS port. I never kept this OS in mind.

Yes, just little different to include OpenGL. This can be more simple, but I didn't compile project with "GL/gl.h". Handling extensions maybe works different on MacOS. I was forced to include OpenGL 3 with all extensions and hide your extensions handling. There are probably better ways

File: src/system_window.cpp (hunk 1 above)
Comment (LifeKILLED) [reply src/system_window.cpp]: Yes. OpenGL 3 is not backward compatible. OpenGL 1 could not be used with extensions. OpenGL 2 is not avaible. Maybe I should have draw quad using DrawArrays or SDL functions. But i decided turn off this way now

**Comment by LifeKILLED** [comment]
I've fixed bug on MacOS. std::atan2 can give negative values. It broke moving and targeting.

46 line in src/messages.cpp fixed all :)

File: src/system_window.cpp (hunk 3 above)
Comment (LifeKILLED) [reply src/system_window.cpp]: fixed

File: src/system_window.cpp (hunk 1 above)
Comment (LifeKILLED) [reply src/system_window.cpp]: fixed

File: src/system_window.cpp (hunk 2 above)
Comment (LifeKILLED) [reply src/system_window.cpp]: fixed

**Comment by Panzerschrek** [comment]
@LifeKILLED, please update `panzer_ogl_lib` submodule to latest master. Than i will merge this PR.

**Comment by LifeKILLED** [comment]
> @LifeKILLED, please update `panzer_ogl_lib` submodule to latest master. Than i will merge this PR.

I didn't commit the submodule and it should be correct without updating

**Comment by Panzerschrek** [comment]
@LifeKILLED , how it can be correct?
As i understand you need both changes in this repo and in `panzer_ogl_lib` to allow MacOS version to compile properly. In order to take changes made in https://github.com/Panzerschrek/panzer_ogl_lib/pull/4 you need to update corresponding submodule in this repo.

**Comment by LifeKILLED** [comment]
> @LifeKILLED , how it can be correct? As i understand you need both changes in this repo and in `panzer_ogl_lib` to allow MacOS version to compile properly. In order to take changes made in [Panzerschrek/panzer_ogl_lib#4](https://github.com/Panzerschrek/panzer_ogl_lib/pull/4) you need to update corresponding submodule in this repo.

Sorry, after rebooting  IDE showed submodule changes, I was update it
